miner/worker.go

```diff
@@ -1208,6 +1210,18 @@ loop:
 	return false, circuitCapacityReached
 }
 
+func (w *worker) checkCurrentTxNum() {
+	txNumInCcc, err := w.circuitCapacityChecker.GetTxNum()
+	if err == nil {
+		txNumInMiner := uint64(w.current.tcount)
+		if txNumInMiner != txNumInCcc {
```

Comment:
I wonder if this scenario is possible:
1. Tracing completes with no error,
2. CCC completes with no error,
3. `ApplyTransaction` fails with an error.
In this case the two transaction counts would get out of sync. But I guess if 3. fails then 1. fails too, is that correct? @HAOYUatHZ

Reply:
yes. you can check `func applyTransaction` in core/state_processor.go

the error can only happen in `fees.CalculateL1DataFee` and `core.ApplyMessage`, which will also be returned in 1, if there's an error